jupyterhub/apihandlers/base.py

```diff
@@ -31,6 +31,12 @@ def check_referer(self):
         if not referer:
             self.log.warning("Blocking API request with no referer")
             return False
+
+        # Checking CORS
+        allow_origin = self.settings.get('allow_origin', None)
+        self.log.warning(allow_origin)
+        if (allow_origin == '*') or (allow_origin and allow_origin in referer_path):
```

Comment:
referer_path isn't defined yet. Plus `in` is an insufficient check for referer_path, which can contain arbitrary strings. Referer will need to be parsed to compare hosts exactly.

For example:

```python
allow_origin = 'https://mydomain.com'
Referer = ''
```